rednotebook/data.py

```diff
@@ -200,41 +200,54 @@ def search(self, text, tags):
                         add_text_to_results = True
             if add_text_to_results:
                 results.append(get_text_with_dots(self.text, 0, TEXT_RESULT_LENGTH))
-        elif text in str(self):
-            # Date contains searched text.
-            results.append(get_text_with_dots(self.text, 0, TEXT_RESULT_LENGTH))
         else:
-            text_result = self.search_in_text(text)
+            for word in words:
+                # Any of the word matches with the date.
+                if word in str(self):
+                    # We don't want to show search results from the text matching
+                    # this date.
+                    words.remove(word)
```

Comment:
list.remove() is dangerous since it uses a linear search in the list. It's better to create a new empty list and add all the non-date words to it. This can be done in a list comprehension: 
``` python
non_date_words = [word for word in words if word not in str(self)]
```

And if word contains "2021" and "2021-04", we only want to add it to the results once. So we can do 

```python
words_contain_date = len(non_date_words) != len(words)
if words_contain_date:
    results.append(get_text_with_dots(self.text, 0, TEXT_RESULT_LENGTH))
```

Reply:
Nice catch Jendrik!